app/src/main/java/com/example/mpvcalculatorrx/mvp/model/CalculatorModel.java

```diff
@@ -0,0 +1,109 @@
+package com.example.mpvcalculatorrx.mvp.model;
+
+import java.util.ArrayList;
+
+public class CalculatorModel extends MvpModel {
+    private ArrayList<Object> buffer;
```

Comment:
Don't use the specific class in declarations, use instead the interfaces, in this case is List.

Reply:
Avoid to use Object in generics, if you would put Long or String instead of Object, you could avoid many casts and instanceof's.